### PR TITLE
Feature/ATG-64 Update board. Checking collision detection. 

### PR DESCRIPTION
**Pull request by IraCherginyshka**

### CODE REVIEW

File: src/app/components/game-board/game-board.component.ts [hunk 1]
```diff
@@ -102,30 +110,56 @@ export class GameBoardComponent implements OnInit, OnDestroy {
...
+        this.boardMatrix = this.currentMatrix;
+        this.setInitialBoardState();
       }
     }, DELAY_FIRST_LEVEL);
   }
 
+  private checkCollisionDetection(step: number, figure: FiguresColors[][]): boolean {
+    let isValid = true;
+    let result: boolean;
+    figure.forEach((line, indexY) => {
```
Comment (oleksandrstarov) [inline src/app/components/game-board/game-board.component.ts]: can we refactor to use Array.every function?

File: src/app/models/figure.model.ts [hunk 2]
```diff
@@ -14,9 +14,24 @@ export class FigureModel {
...
 
     figureMatrix.forEach((line, index) => {
-      const targetLine = [...boardMatrix[index + height]];
-      targetLine.splice(figurePosition, line.length, ...line);
-      matrixWithFigure[index + height] = [...targetLine];
+      let activePosition = 0;
+      let hasActiveBlock = false;
+      const activeBlocksLine = line.filter((block, ind) => {
+        if (block !== FiguresColors.DEFAULT && !hasActiveBlock) {
+          activePosition = ind;
```
Comment (oleksandrstarov) [inline src/app/models/figure.model.ts]: what does ind means?

**Review by oleksandrstarov — CHANGES_REQUESTED** [review]

File: src/app/components/game-board/game-board.component.ts (hunk 1 above)
Comment (IraCherginyshka) [reply src/app/components/game-board/game-board.component.ts]: Yes
Changed

File: src/app/models/figure.model.ts (hunk 2 above)
Comment (IraCherginyshka) [reply src/app/models/figure.model.ts]: I changed name

File: src/app/components/game-board/game-board.component.ts [hunk 3]
```diff
@@ -100,23 +109,41 @@ export class GameBoardComponent implements OnInit, OnDestroy {
...
+      return line.every((block, indexX) => {
+        if (block !== FiguresColors.DEFAULT) {
+          const x = this.figurePosition + indexX + step;
+          const y = this.lineWithFigure + indexY;
+          if (x >= 0 && x <= QUANTITY_BLOCKS_WIDTH && y < QUANTITY_BLOCKS_HEIGHT) {
+            return this.boardMatrix[y][x] === FiguresColors.DEFAULT;
+          }
+          return false;
+        }
+        return true;
```
Comment (oleksandrstarov) [inline src/app/components/game-board/game-board.component.ts]: can we avoid code foldings of if by using such approach

`if (block === FiguresColors.DEFAULT) {
         return true;
}
const x = this.figurePosition + indexX + step;
const y = this.lineWithFigure + indexY;
if (x >= 0 && x <= QUANTITY_BLOCKS_WIDTH && y < QUANTITY_BLOCKS_HEIGHT) {
           return this.boardMatrix[y][x] === FiguresColors.DEFAULT;
 }
 return false;
`

**Review by oleksandrstarov — CHANGES_REQUESTED** [review]

File: src/app/components/game-board/game-board.component.ts (hunk 3 above)
Comment (IraCherginyshka) [reply src/app/components/game-board/game-board.component.ts]: Changed

**Review by oleksandrstarov — APPROVED** [review]

**Pull request merged**